Release new flathub maintenance version

nbenitez opened the pull request:
- Fix app crashing if built with GCC >= 13.1 
 - Update runtime to 24.08 and deps to latest versions
 - Update gtkmm.json versions
 - Remove "Pictures" folder permission
 - Add wayland socket pemission
 - Re-phrase metainfo file Summary according to guidelines
 - Update screenshots to improve them and also fix this [flathub page issue](https://github.com/flathub-infra/website/issues/4192)  


<summary>Fix app crashing if built with GCC >= 13.1</summary>
<details>

App worked fine in runtime 22.08 (GCC 12.2.0)
but starting crashing with runtime 23.08 which
comes with GCC 13.1.0

After debugging the crash, the problem was the
order in which the class member initializers were
executed (more info [here](https://stackoverflow.com/a/1828058)) because the code
had some classes initializing `Gtk::RadioButton`
members with a reference to a `Gtk::RadioButtonGroup`
that was still not initialized, because that
`Gtk::RadioButtonGroup` member was declared in the
header file after the `Gtk::RadioButton` in question.

Apparently GCC 12 was more resilient to this kind
of inconsistency.
</details>

hfiguiere commented on gtkmm.json:
```diff
@@ -64,9 +73,9 @@
+                    "sha256": "38684cff317273615c67b8fa9806f16299d51e5506d9b909bae15b589fa99cb6",
                     "type": "archive",
-                    "url": "https://ftp.gnome.org/pub/GNOME/sources/glibmm/2.66/glibmm-2.66.0.tar.xz"
+                    "url": "https://download.gnome.org/sources/glibmm/2.82/glibmm-2.82.0.tar.xz"
```
not sure why you try to blindly upgrade everything to the latest, but glibmm 2.66 is the last version that is compatible with gtkmm 3.24. 2.68 broke API and is only for gtkmm4.

Latest is not always best.

nbenitez replied:
hey Hubert, thanks for your guidance, I'm not savvy in gtkmm bindings release policy and I wrongly thought I could update to latest. Sooner or later I would have found out the build won't succeed, but thank you for letting me know in advance.

hfiguiere commented on gtkmm.json:
```diff
@@ -2,9 +2,18 @@
+                "sed -i 's/cairomm-1.0/cairomm-1.16/' meson.build",
+                "sed -i 's/pangomm-1.4/pangomm-2.48/' meson.build",
+                "sed -i 's/atkmm-1.6/atkmm-2.36/' meson.build"
+            ]
```
no no and no. You are trying to build with incompatible version here. There is a reason why they have a different name. It because they are NO LONGER API compatible.

nbenitez replied:
Ok, I will resume this work in the weekend.

hfiguiere commented on gtkmm.json:
```diff
@@ -45,9 +45,9 @@
+                    "sha256": "c3d23b37dfd6e39f2e09f091b77b1541fbfa17c4f0b6bf5c89baef7229080e17",
                     "type": "archive",
-                    "url": "https://ftp.gnome.org/pub/GNOME/sources/libsigc++/2.10/libsigc++-2.10.6.tar.xz"
+                    "url": "https://download.gnome.org/sources/libsigc++/3.6/libsigc++-3.6.0.tar.xz"
```
not sure why you insist on upgrading this with an incompatible version.

hfiguiere replied:
```
../meson.build:106:13: ERROR: Dependency "sigc++-2.0" not found, tried pkgconfig and cmake
```

I mean the local build would fail the same way.

nbenitez replied:
Man is the only animal that trips twice over the same stone! 🙂

hfiguiere replied:
… on the same stone man put in front to trip on.

nbenitez committed:
Update runtime to 24.08 and deps to latest versions

 - Update gtkmm.json versions
 - Remove "Pictures" folder permission
 - Add wayland socket pemission
 - Re-phrase metainfo file Summary according to guidelines
 - Update screenshots

nbenitez committed:
Fix app crashing if built with GCC >= 13.1

App worked fine in runtime 22.08 (GCC 12.2.0)
but starting crashing with runtime 23.08 which
comes with GCC 13.1.0

After debugging the crash, the problem was the
order in which the class member initializers were
executed (more info here[1]) because the code
had some classes initializing Gtk::RadioButton
members with a reference to a Gtk::RadioButtonGroup
that was still not initialized, because that
Gtk::RadioButtonGroup member was declared in the
header file after the Gtk::RadioButton in question.

Apparently GCC 12 was more resilient to this kind
of inconsistency.

[1] https://stackoverflow.com/a/1828058

nbenitez committed:
Release new flathub maintenance version

nbenitez commented:
This is now finished, @hfiguiere  let me know if you have any comment otherwise I'll commit in a couple days.

hfiguiere commented:
my only comment was about the multiple broken builds.... it's your package.
